### PR TITLE
#2894 - Bug - Mass offering change cannot be approved

### PR DESCRIPTION
**Pull request by lewischen-aot**
- Ensured that application.currentAssessment is populated. 
- Updated the application for the current assessment in the else logic.
- Created E2E test to check in-progress application was canceled.

### CODE REVIEW

File: sources/packages/backend/apps/api/src/services/education-program-offering/education-program-offering.service.ts
Comment (dheepak-aot) [reply sources/packages/backend/apps/api/src/services/education-program-offering/education-program-offering.service.ts:1177]: As a solution I would advise to remove the alias `assessmentId`. And also remove the property from the model `ApplicationAssessmentSummary`

image

File: sources/packages/backend/apps/api/src/route-controllers/education-program-offering/_tests_/e2e/education-program-offering.aest.controller.assessOfferingChangeRequest.e2e-spec.ts
Comment (dheepak-aot) [inline sources/packages/backend/apps/api/src/route-controllers/education-program-offering/_tests_/e2e/education-program-offering.aest.controller.assessOfferingChangeRequest.e2e-spec.ts:314]: I couldn't follow the test description here.

The test is validating 
the offering change for an offering with one or more applications not in completed status being approved.

File: sources/packages/backend/apps/api/src/route-controllers/education-program-offering/_tests_/e2e/education-program-offering.aest.controller.assessOfferingChangeRequest.e2e-spec.ts
Comment (dheepak-aot) [inline sources/packages/backend/apps/api/src/route-controllers/education-program-offering/_tests_/e2e/education-program-offering.aest.controller.assessOfferingChangeRequest.e2e-spec.ts:331]: I am not sure if creating an array of applications adds value to the test.

I would say either create an application in `In-Progress` (or `Assessment` or `Enrolment Status`) and one Completed application both.

File: sources/packages/backend/apps/api/src/route-controllers/education-program-offering/_tests_/e2e/education-program-offering.aest.controller.assessOfferingChangeRequest.e2e-spec.ts
Comment (dheepak-aot) [inline sources/packages/backend/apps/api/src/route-controllers/education-program-offering/_tests_/e2e/education-program-offering.aest.controller.assessOfferingChangeRequest.e2e-spec.ts:372]: Please check the status of preceding and requested offering as well.

**Comment by dheepak-aot** [comment]
Please have a look at the comments.

File: sources/packages/backend/apps/api/src/services/education-program-offering/education-program-offering.service.ts
Comment (dheepak-aot) [inline sources/packages/backend/apps/api/src/services/education-program-offering/education-program-offering.service.ts:1174]: Please remove the `assessmentId` here as well.

File: sources/packages/backend/apps/api/src/route-controllers/education-program-offering/_tests_/e2e/education-program-offering.aest.controller.assessOfferingChangeRequest.e2e-spec.ts
Comment (dheepak-aot) [inline sources/packages/backend/apps/api/src/route-controllers/education-program-offering/_tests_/e2e/education-program-offering.aest.controller.assessOfferingChangeRequest.e2e-spec.ts:318]: Please remove the blank line.

File: sources/packages/backend/apps/api/src/route-controllers/education-program-offering/_tests_/e2e/education-program-offering.aest.controller.assessOfferingChangeRequest.e2e-spec.ts
Comment (dheepak-aot) [inline sources/packages/backend/apps/api/src/route-controllers/education-program-offering/_tests_/e2e/education-program-offering.aest.controller.assessOfferingChangeRequest.e2e-spec.ts:324]: Minor. it can be applicationId array to use directly in query.

File: sources/packages/backend/apps/api/src/route-controllers/education-program-offering/_tests_/e2e/education-program-offering.aest.controller.assessOfferingChangeRequest.e2e-spec.ts
Comment (dheepak-aot) [inline sources/packages/backend/apps/api/src/route-controllers/education-program-offering/_tests_/e2e/education-program-offering.aest.controller.assessOfferingChangeRequest.e2e-spec.ts:314]: can the description be like

`Should approve the offering change when the offering has one or more applications not in Completed status and add a institution note on approval.`

Comment (lewischen-aot) [reply sources/packages/backend/apps/api/src/route-controllers/education-program-offering/_tests_/e2e/education-program-offering.aest.controller.assessOfferingChangeRequest.e2e-spec.ts]: Sure. The description is updated.

File: sources/packages/backend/apps/api/src/route-controllers/education-program-offering/_tests_/e2e/education-program-offering.aest.controller.assessOfferingChangeRequest.e2e-spec.ts
Comment (lewischen-aot) [reply sources/packages/backend/apps/api/src/route-controllers/education-program-offering/_tests_/e2e/education-program-offering.aest.controller.assessOfferingChangeRequest.e2e-spec.ts]: Done.

File: sources/packages/backend/apps/api/src/route-controllers/education-program-offering/_tests_/e2e/education-program-offering.aest.controller.assessOfferingChangeRequest.e2e-spec.ts
Comment (dheepak-aot) [inline sources/packages/backend/apps/api/src/route-controllers/education-program-offering/_tests_/e2e/education-program-offering.aest.controller.assessOfferingChangeRequest.e2e-spec.ts:287]: Even though it is existing logic, I would recommend to get the applications based in applicationIds like the recent test. Other wise this test will not be independent by itself.

**Review by dheepak-aot — APPROVED** [review]
Thanks for doing the changes. Looks good 👍

Added 2 minor comments on E2E tests.

**Pull request merged**
